### ISSUE TITLE
Add Spectrum Field Label Pattern

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#475]
### Expected Behaviour
https://opensource.adobe.com/spectrum-css/components/fieldlabel/ should be available in the library.

### Actual Behaviour
Missing

### The future
![image](https://user-images.githubusercontent.com/1156657/73952567-1a012680-48cd-11ea-9d97-6ef1e611ba97.png)

**Comment by Westbrook** [comment]
I _think_ it will be good to structure this pattern in a way that is similar to this: https://stackblitz.com/edit/a11y-input?file=my-element.js Possibly a little more like a mixin than a base class, though I'm not sure. The reasoning here is that the `label` and the `input` element need to exist in the same DOM tree in order to be attached in an accessible manner. This means that you can't use a `<sp-label>Input Label</sp-label><sp-textfield></sp-textfield>` style pattern as the subsequent `label` and `input` would exist in separate DOM trees.

**Comment by Westbrook** [comment]
I'm looking into a new version of this that leverages the idea that we move form elements into the light DOM of their containing elements, and then resolve them in scope with JS and dynamically associate those form elements with an `<sp-field-label>` element. It's currently looking promising, but I need to expand the pattern beyond just `<sp-textfield>` form elements to ensure that this is something that can be recommended as a library wide solutions for the need to be able to label form elements.

**Comment by Westbrook** [comment]
Released.
